### ISSUE TITLE
Required version not being reported properly during upgrade

### ISSUE THREAD

**Issue by 1DontEx1st** [custom-components/hacs#557]
## Version of HACS

0.14.2

## Describe the bug
I'm trying to upgrade a plugin, specifically Button Entity Row, and am getting the following error:
![image](https://user-images.githubusercontent.com/6410132/65381709-f5f9cb80-dcc4-11e9-8b3c-148d68ff698e.png)
I'm currently running HA 0.99.1, as per the screenshot, but the plugin requires 0.99.2 - not "None" as is being reported by HACS.

## Debug log

```text
2019-09-22 03:01:01 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-09-22 03:01:02 INFO (MainThread) [homeassistant.setup] Setting up hacs
2019-09-22 03:01:02 DEBUG (MainThread) [hacs] Configuration type: yaml
2019-09-22 03:01:02 INFO (MainThread) [hacs]
https://github.com/custom-components/hacs/issues
2019-09-22 03:01:02 DEBUG (MainThread) [hacs.repository.integration.custom-components.hacs] Checking repository.
2019-09-22 03:01:06 INFO (MainThread) [hacs.repository.integration.custom-components.hacs] Registration complete
2019-09-22 03:01:06 INFO (MainThread) [hacs.data] Restore started
2019-09-22 03:01:06 INFO (MainThread) [hacs.data] Restore done
2019-09-22 03:01:06 INFO (MainThread) [homeassistant.setup] Setup of domain hacs took 4.1 seconds.
2019-09-22 03:01:14 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.hacs
2019-09-22 03:01:14 DEBUG (MainThread) [hacs] 4952
2019-09-22 03:01:14 DEBUG (MainThread) [hacs] 2019-09-22 04:01:05
2019-09-22 03:01:14 INFO (MainThread) [hacs] Loading known repositories
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.data] Saving data
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.backup] Backup for /config/.storage/hacs.hacs, created in /tmp/hacs_backup/hacs.hacs
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.backup] Backup dir /tmp/hacs_backup/ cleared
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.backup] Backup for /config/.storage/hacs.installed, created in /tmp/hacs_backup/hacs.installed
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.backup] Backup dir /tmp/hacs_backup/ cleared
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.backup] Backup for /config/.storage/hacs.repositories, created in /tmp/hacs_backup/hacs.repositories
2019-09-22 03:01:17 DEBUG (MainThread) [hacs.backup] Backup dir /tmp/hacs_backup/ cleared
2019-09-22 03:01:20 DEBUG (MainThread) [hacs.http] Endpoint (overview) called
2019-09-22 03:01:31 DEBUG (MainThread) [hacs.http] Endpoint (repository) called
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.data] Saving data
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.backup] Backup for /config/.storage/hacs.hacs, created in /tmp/hacs_backup/hacs.hacs
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.backup] Backup dir /tmp/hacs_backup/ cleared
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.backup] Backup for /config/.storage/hacs.installed, created in /tmp/hacs_backup/hacs.installed
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.backup] Backup dir /tmp/hacs_backup/ cleared
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.backup] Backup for /config/.storage/hacs.repositories, created in /tmp/hacs_backup/hacs.repositories
2019-09-22 03:01:33 DEBUG (MainThread) [hacs.backup] Backup dir /tmp/hacs_backup/ cleared
2019-09-22 03:02:52 DEBUG (MainThread) [hacs.http] Endpoint (settings) called
2019-09-22 03:03:03 DEBUG (MainThread) [hacs.http] Endpoint (settings) called
2019-09-22 03:06:50 DEBUG (MainThread) [hacs.http] Endpoint (overview) called
2019-09-22 03:06:54 DEBUG (MainThread) [hacs.http] Endpoint (settings) called
2019-09-22 03:07:01 DEBUG (MainThread) [hacs.http] Endpoint (store) called
2019-09-22 03:07:18 DEBUG (MainThread) [hacs.http] Endpoint (overview) called
2019-09-22 03:07:34 DEBUG (MainThread) [hacs.http] Endpoint (settings) called
2019-09-22 03:08:58 DEBUG (MainThread) [hacs.http] Endpoint (overview) called
2019-09-22 03:09:01 DEBUG (MainThread) [hacs.http] Endpoint (repository) called
```

**Comment by aptonline** [comment]
I’m seeing the same issue with the same component. I noticed that button entry row supports the new hacs.json file, is this needed to report correctly version upgrades etc?

**Comment by ludeeus** [comment]
Yes, and that part is working (Since you get a popup), but the handling in that popup is wrong.

**Comment by aptonline** [comment]
Where does the fix lay with the developer or HACS?

**Comment by ludeeus** [comment]
HACS

**Comment by ludeeus** [comment]
fixed in 0.15.2
